candle-onnx/src/eval.rs

```diff
@@ -1960,6 +1960,71 @@ fn simple_eval_(
                 let output = input.sign()?;
                 values.insert(node.output[0].clone(), output);
             }
+            "Resize" => {
+                let input = get(&node.input[0])?;
+
+                let scales = if node.input.len() > 2 && !node.input[2].is_empty() {
+                    Some(get(&node.input[2])?)
+                } else {
+                    None
+                };
+
+                let sizes = if node.input.len() > 3 && !node.input[3].is_empty() {
+                    Some(get(&node.input[3])?)
+                } else {
+                    None
+                };
+
+                if scales.is_none() && sizes.is_none() {
+                    bail!("Either scales or sizes must be provided for Resize operation");
+                }
+
+                // Get attributes with their default values according to ONNX spec
+                let mode = get_attr_opt::<str>(node, "mode")?.unwrap_or("nearest");
+                let coordinate_transformation_mode =
+                    get_attr_opt::<str>(node, "coordinate_transformation_mode")?
+                        .unwrap_or("half_pixel");
+
+                let output_dims = if let Some(sizes_tensor) = sizes {
+                    // Use specified sizes
+                    sizes_tensor
+                        .to_vec1::<i64>()?
+                        .iter()
+                        .map(|&d| d as usize)
+                        .collect::<Vec<_>>()
+                } else if let Some(scales_tensor) = scales {
+                    // Calculate from scales
+                    let scale_values = scales_tensor.to_vec1::<f32>()?;
+                    input.dims()
+                        .iter()
+                        .enumerate()
+                        .map(|(i, &d)| (d as f32 * scale_values[i]) as usize)
+                        .collect::<Vec<_>>()
+                } else {
+                    unreachable!("Either scales or sizes should be present");
```

Comment:
I would use `bail!` rather than unreachable here just in case, we do the check with `is_none()` above but it's possible that the check will be removed through some refactoring. Actually I would even remove the check above so that it's entirely handled here.
Also what about the case where both `scales` and `sizes` are set, should we have an error or is there a reason to choose `sizes` first?

Reply:
Yeah, It should be an error: according to [onnx documentation](https://onnx.ai/onnx/operators/onnx__Resize.html#inputs):
> One of ‘scales’ and ‘sizes’ MUST be specified and it is an error if both are specified.